### PR TITLE
rose_prune: support custom cycle format

### LINKED ISSUE #1645: rose_prune: flexible date-time format for cycle formatting string
We should support flexible date-time format for the `%(cycle)s` field in a formatting string in the configuration of `rose_prune`.

### PR DESCRIPTION
**Pull request by matthewrmshin**
Close #1645.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@arjclark @kaday please review.

File: doc/rose-rug-task-run.html [hunk 1]
```diff
@@ -896,6 +896,17 @@ <h2 id="rose-task-run.util.rose_prune">Built-in Application:
     <p>The following settings are accepted:</p>
 
     <dl>
+      <dt><kbd>cycle-format{key}=format</kbd></dt>
+
+      <dd>Specify a key to a format string for use in conjunction with a
+      <kbd>prune{item-root}=cycle:globs ...</kbd> setting. For example, we may
+      have something like <kbd>cycle-format{cycle_year}=CCYY</kbd> and
+      <kbd>prune{share}=-P1Y:xmas-present-%(cycle_year)s/</kbd>. The
```
Comment (kaday) [inline doc/rose-rug-task-run.html]: The other examples then follow the example with what the example would clear out.

**Comment by kaday** [comment]
Other than that one comment it looks okay and passes the test battery in my environment.

**Comment by arjclark** [comment]
Looks ok to me. Test battery passing in my environment. Fine to merge from my point of view once @kaday's comment has been addressed.

File: doc/rose-rug-task-run.html (hunk 1 above)
Comment (matthewrmshin) [inline doc/rose-rug-task-run.html]: Done.

**Comment by kaday** [comment]
Thanks.

**Pull request merged**